source/common/network/connection_impl.cc

```diff
@@ -715,6 +715,10 @@ void ConnectionImpl::onWriteReady() {
     // It is possible (though unlikely) for the connection to have already been closed during the
     // write callback. This can happen if we manage to complete the SSL handshake in the write
     // callback, raise a connected event, and close the connection.
+    if (transport_socket_->ssl() != nullptr &&
+        transport_socket_->ssl()->sslSyscallErrorOccurred()) {
+      setFailureReason(absl::StrCat("The connection has been closed during the write callback."));
```

Comment:
`The connection has been closed during the write callback` isn't a reason that the end user can easily understand. Can you make this more descriptive regarding things that may cause this to happen?

Reply:
Here is the generic prefix for failure reason, more descriptive info comes from underlying SSL lib via `transport_socket_->failureReason` when the final error message is constructed [here](https://github.com/envoyproxy/envoy/blob/020de5c9838729acc007b4748ecae63331182c5b/source/common/network/connection_impl.cc#L560) . An example in this PR is that `KEY_USAGE_BIT_INCORRECT` is appended in the final error message.  

If you still feel it is not descriptive enough, I am wonder do you have suggested description?

Reply:
This error gets surfaced in access logs (and other places), but this string is specific to the implementation and isn't easily understood by users.